# 3. Protein Structure Prediction

Since a protein's entire role in the body is controlled by its structure, researchers have been studying the protein folding process and trying to simulate it computationally. Through such simulations, scientists can not only better understand what impacts the folding and structure of a protein, but also fold de novo (new) proteins and test their capabilities computationally before using resources in the wet lab. 

## Protein Folding Programs
  - [**AlphaFold2**](https://github.com/alyssablood/Perez-Lab-Intro-Tutorials/blob/main/5.%20Protein%20Interactions.md#alphafold-competitive-binding-assays-af-cba): Extremely popular tool for deriving a 3D structure from an amino acid sequence. Very high accuracy, industry standard. You can read the AlphaFold official paper [here](https://www.nature.com/articles/s41586-021-03819-2).
  - [**AlphaFold3**](https://github.com/alyssablood/Perez-Lab-Intro-Tutorials/blob/main/5.%20Protein%20Interactions.md#alphafold-competitive-binding-assays-af-cba): Expansion of AlphaFold2 that also analyzes interacions between molecules. This program won 2024 Nobel Prize in Chemistry. You can read the AlphaFold3 official paper [here](https://www.nature.com/articles/s41586-024-07487-w).
  - **RoseTTAFold**: Similar to AlphaFold, RoseTTAFold uses Multiple Sequence Alignments (MSAs) and templates for high accuracy. You can read the AlphaFold3 official paper [here](https://www.ipd.uw.edu/wp-content/uploads/2021/07/Baek_etal_Science2021_RoseTTAFold.pdf).
  - **ESMFold**: Unlike AlphaFold and RoseTTAFold, ESMFold prioritizes speed and scalability by avoiding MSAs and utilizing the internal knowledge of large protein language models trained on vast protein sequence data. It is excellent for exploring vast sequence spaces and producing many models quickly. You can read the ESMFold official paper [here](https://www.science.org/doi/abs/10.1126/science.ade2574).
  - [**AMBER**](https://github.com/alyssablood/Perez-Lab-Intro-Tutorials/blob/main/6.%20Protein%20Dynamics.md#beyond-static-folds-ensembles-and-motion): Uses physics-based (p-chem) methods to apply [force fields](https://en.wikipedia.org/wiki/Force_field_(chemistry)) (vector equations) to molecules in order to predict folding, docking, and general behavior. Good for very specific results, binding energy and other outputs are considered reliable. AMBER's homepage and tutorials are linked [here](https://ambermd.org/index.php).

## How Do They Work 

While each program works differently, the concept of predicting protein structures can best be seen through the AlphaFold architecture -
![alphafold](./images/alphafold.png)

Like all neural-network-based AI programs, AlphaFold has an **Input**(protein sequence), a **Hidden Layer** where all the folding takes place, and an **Output** (protein structure).

In the Hidden Layer, the amino acid sequence is compared to other amino acid sequences in the Protein Data Bank (PDB), and a **multiple sequence alignment (MSA)** is performed to see what amino acids are conserved (the same across multiple sequences) as well as the general patterns displayed. At the same time, the input amino acid sequence is also searched for structures in the PDB, and the same alignment method is used to compare similar structures and find patterns. This is shown in the "B. database search" in the image above. These patterns are then fed to the Evoformer, combined, and used to generate an initial structure, as shown in the "C. prediction model" in the image above. However, this structure is very poor and usually doesn't represent a structure that would exist in the real world. So, the prediction process is recycled three times, yielding the final protein structure.

## Alphafold Tutorials 

### Graphical User Interface (GUI) Tutorial with AlphaFold Server

AlphaFold has a server to easily run jobs with a graphical interface as a user-friendly alternative to running the program in terminal. 

Go to https://alphafoldserver.com/welcome and click continue with google

You can read the information at the top of the screen, but the input section looks like this - 

 ![af_server](./images/af_server.png)

In the *Input* box, you can paste the amino acid sequence of the protein you are trying to fold. 

For this tutorial, you can click on one of AlphaFold’s examples, Protein-Glycan-Ion: PDB 7BBV. Notice how after clicking the button the information auto-fills the input bar. In this example, there are 2 entities - the protein and the Ca Ion. By typing the PDB code 7BBV into the PDB (https://www.rcsb.org/) we can see that the protein is a Pectate lyase, which is a protein that degrades complex polysaccharides and major components of the plant primary cell wall, mainly used in the food industry. The PDB page for this protein also shows what its predicted structure should look like, as solved by X-ray diffraction. 

 ![af_pdb](./images/af_pdb.png)

Hit the “continue and preview job” button and then “confirm and submit job”. The job should run very quickly since it is an AlphaFold example

Click on the name of your job at the bottom of the screen in the table  - “Protein Protein-Glycan-Ion: PDB 7BBV.” 

You should now see the AlphaFold predicted structure of the protein. Notice the confidence bars above the model. These are measured in **plDDT** ( predicted Local Distance Difference Test). Since the model is mostly colored in blue, AlphaFold is very confident in this structure. This can be seen by visually comparing the AlphaFold output with the experimental structure in the PDB. They are highly similar, except for a random, low-confidence helix jutting out from the main protein. Note that your protein structure may look slightly different from mine, as AlphaFold incorporates random elements into its architecture and input processing. 

![af_output](./images/af_output.png)


But how different are these two models? 


To figure this out, we must perform a **Structural Superposition** with visualization software. For this tutorial, I will choose ChimeraX. 

First, hit the download button above the AlphaFold output model and download the files to a preferred location. Then, extract the files. 

Open up ChimeraX and open the model examplefold_pdb_7bbv_model_4 (AlphaFold always produces 5 total, so we’ll randomly choose the last one). 

>File → Open → examplefold_pdb_7bbv_model_4

You should now see the model shown on the AlphaFold server

Open the Command Line Interface if it isn’t there already

>Tools → Command Line Interface

Open the experimental PDB model 7bbv through the command line and hit enter

> open 7bbv

You should now see 2 differently colored models in ChimeraX (the models sidebar labels the 2 models) 

Perform a structural alignment

>Tools → Structural Analysis → Matchmaker

In the menu, select the bullet point: Best-aligning pair of chains between reference and match structure

This aligns the structure of one model to the reference structure, which will be the experimental 7bbv (#2) model. Since there are only 2 models, it does not matter which model is in the reference structure or structure(s) to match boxes at the top of the menu, but this would be important if there were more. 

Click the 'Apply' button below, and then click 'OK'. You should now see the two models overlaid on top of each other. You can now clearly see which regions of protein are the same (overlapping of blue and tan) and which regions are different. 

![af_msa](./images/af_msa.png)

### Command Line Interface (CLI) Tutorial with AlphaFold through HiPerGator
For the vast majority of research projects, AlphaFold is typically used from the command line, as it offers researchers more control over settings and the opportunity to produce results of higher quality and quantity. 

First, open the command line interface on your device (for example, PowerShell for Windows)

SSH to HiPerGator and complete the sign-in process -

>ssh “username”@hpg.rc.ufl.edu

Once in the login node/home directory, navigate the **Orange** directory -

> cd /orange/”HiPerGator_Professor_sponsor”/”username”/

Navigate to a place you would like to install AlphaFold. Upon installation, a directory gets created for you. 

Go to the Github page for AlphaFold - https://github.com/google-deepmind/alphafold3 . We will be installing AlphaFold 3 since it is the most advanced version of the software. 

Click on the green box that says “code” and copy the URL above “Clone using the web URL.” 

Switch back to your command line interface and once you are in the location you want, clone the AlphaFold 3 GitHub repository - 

> git clone https://github.com/google-deepmind/alphafold3.git 

Read through this document on the GitHub repository for further installation instructions - https://github.com/google-deepmind/alphafold3/blob/main/docs/installation.md 

Once AlphaFold3 is fully installed in your HiPerGator Orange Storage, look for an input JSON file named fold_input.json. Open the file -
```
vi fold_input.json 
```
It should look something like this - 

```
{
  "name": "2PV7",
  "sequences": [
    {
      "protein": {
        "id": ["A", "B"],
        "sequence": "GMRESYANENQFGFKTINSDIHKIVIVGGYGKLGGLFARYLRASGYPISILDREDWAVAESILANADVVIVSVPINLTLETIERLKPYLTENMLLADLTSVKREPLAKMLEVHTGAVLGLHPMFGADIASMAKQVVVRCDGRFPERYEWLLEQIQIWGAKIYQTNATEHDHNMTYIQALRHFSTFANGLHLSKQPINLANLLALSSPIYRLELAMIGRLFAQDAELYADIIMDKSENLAVIETLKQTYDEALTFFENNDRQGFIDAFHKVRDWFGDYSEQFLKESRQLLQQANDLKQG"
      }
    }
  ],
  "modelSeeds": [1],
  "dialect": "alphafold3",
  "version": 1
}
```
Although it may look intimidating, notice the resemblance to the GUI version on the AlphaFold server.
1. "name": "2PV7"
This is a user-defined label for the job.
It is often named after a PDB (Protein Data Bank) ID, but it can be any string.
AlphaFold uses this name to label the output files (e.g., 2PV7_summary_confidences_0.json).
2. "sequences": [...]
This is the core of the input. It is a list containing all the "entities" (proteins, DNA, RNA, or ligands) you want to include in the prediction.
"protein": { ... }: This specifies that the entity is a protein made of amino acids.
"id": ["A", "B"]: This is the chain identifier for this specific molecule. For example, if you have 3 proteins in this AlphaFold job and label them each A, B, and C, then the output PDB file will label these chains the same way. Since this example has A and B for one molecule, this indicates it is a homodimer (two identical copies of the same protein interacting with each other). 
"sequence": "GMR...QG": This is the amino acid sequence of the protein usually obtained from the PDB.
While this example only has a protein as an input, you can easily represent a sugar ligand, for example, with -
"ligand": {
"id": "A",
 "smiles": "C[C@H]1…O)O)O"
The “smiles” is the “Simplified Molecular Input Line Entry System,” which is used to translate a ligand’s three-dimensional structure into a string of symbols that is easily understood by AlphaFold and other programs.

3. "modelSeeds": [1]
The "seed" controls the random number generator used during the prediction process.
[1]: This tells the model to run one prediction using the random seed 1. This is the recommended default. 
If you wanted to see different variations or "clusters" of how the protein might fold, you could provide multiple seeds (e.g., [1, 2, 3]), and the model would generate three independent structures.
4. "dialect": "alphafold3"
This specifies the JSON schema being used. Because different versions of AlphaFold or related tools (like RoseTTAFold) might use similar formats, this ensures the software interprets the keys (like sequences or modelSeeds) correctly according to AF3's specific requirements.
5. "version": 1
This refers to the version of the input format itself. If Google DeepMind updates the AlphaFold 3 interface or adds new features, this version number helps ensure backward compatibility with older input files.
As mentioned earlier, this input highly resembles that of the AlphaFold server input, but offers a lot more control over the settings of the program. 

To see if AlphaFold through the CLI models a better structure than AlphaFold through the GUI Server, change the name, ID, and sequence to match that of 7bbv from the previous tutorial. 

After filling out the input file, running it is not as simple as clicking a button. We must ensure that AlphaFold is using HiPerGator’s resources to run efficiently. To do this, we must tap into HiPerGator’s Slurm system.

In whatever location you prefer inside the AlphaFold directory, create a Slurm script (ends in .sh or .slurm - 
> vi af3.slurm

Using the information from the Core Technical Foundations tutorial, write the Slurm script. For example, it may look like this - 

```
 #!/bin/bash
#SBATCH --job-name=test_run   	# Job name
#SBATCH --ntasks=1               	# Number of MPI ranks
##SBATCH --cpus-per-gpu=1           	# Number of cores per MPI rank
##SBATCH --gpus-per-task=1
##SBATCH --ntasks-per-node=1
##SBATCH --gpus-per-node=1
#SBATCH --gpus=1


##SBATCH --nodes=1                 	 # Number of nodes
#SBATCH --mem-per-cpu=20gb         	 # Memory per processor
#SBATCH --time=168:00:00             	 # Time limit hrs:min:sec
#SBATCH --output=af3.log    		 # Standard output and error log
#SBATCH --mail-type=ALL
#SBATCH --mail-user=your_email
pwd; hostname; date

module load ufrc cuda/12 gcc hhsuite

singularity exec \
     --nv \
     /orange/”HiPerGator_Professor_sponsor”/”username”/alphafold3/alphafold3.sif \
     python/orange/”HiPerGator_Professor_sponsor”/”username”/alphafold3/run_alphafold.py \

     --json_path=/orange/”HiPerGator_Professor_sponsor”/”username”/af_input/correct_fold_input.json \
     --model_dir=/orange/”HiPerGator_Professor_sponsor”/”username”/alphafold3/models \
     --db_dir=/orange/”HiPerGator_Professor_sponsor”/”username”/alphafold3/db \
     --output_dir=/orange/”HiPerGator_Professor_sponsor”/”username”/alphafold3/test_run/
```
After configuring the Slurm Script, save and exit

Run the Slurm Script and wait for results, and anticipate troubleshooting -

>sbatch af3.slurm





### [Docking](https://github.com/alyssablood/Perez-Lab-Intro-Tutorials/blob/main/5.%20Protein%20Interactions.md#docking-methods)
  - Docking programs can be very similar to protein dynamics, however they are not capable of predicting mechanisms as they do not consider time-dependant factors. They are used primarily to predict final docked conformations.
  - [**AutoDock**](https://github.com/alyssablood/Perez-Lab-Intro-Tutorials/blob/main/5.%20Protein%20Interactions.md#docking-methods): Predicts how ligand and protein interact, and what binding conformation would occur. P-chem methods are NOT used, therefore specific numbers are considered unreliable. Good for preliminary testing to get shape and quantity of docked poses. An overview of the AutoDock suite's capabilities can be found [here](https://pmc.ncbi.nlm.nih.gov/articles/PMC4868550/).

## [Protein Design](https://github.com/alyssablood/Perez-Lab-Intro-Tutorials/blob/main/4.%20Protein%20Design.md#4-protein-design)
  - [**BindCraft**](https://github.com/alyssablood/Perez-Lab-Intro-Tutorials/blob/main/4.%20Protein%20Design.md#bindcraft-and-other-emerging-methods): Good for desiging a new binder custom fit to a target protein. It is generally very accurate, but follows strict garbage in garbage out policy. Input protein must be high accuracy, and runs take a relatively long time. You can read the BindCraft official paper [here](https://www.nature.com/articles/s41586-025-09429-6). Click [here](https://github.com/alyssablood/Perez-Lab-Intro-Tutorials/blob/main/8.%20Community%20Challenges%3A%20unbiased%20assessment%20of%20methods.md#igem-projects-on-campus) to see how UF's undergraduate synthetic biology team (iGEM) is used BindCraft for deep space travel research!
  - [**ProteinMPNN**](https://github.com/alyssablood/Perez-Lab-Intro-Tutorials/blob/main/4.%20Protein%20Design.md#rfdiffusion-and-proteinmpnn): Optimizes amino acid sequences for most stable protein backbones. Good for designing new target proteins. You can read a paper introducing ProteinMPNN [here](https://www.science.org/doi/10.1126/science.add2187).
  - [**RFDiffusion**](https://github.com/alyssablood/Perez-Lab-Intro-Tutorials/blob/main/4.%20Protein%20Design.md#rfdiffusion-and-proteinmpnn): Generates new protein backbones using diffusion models. Adds in noise to allow for flexibility in folding. Results MUST be checked carefully. It is very easy to add too much noise and generate garbage outputs. You can read RFDiffusion's official paper [here](https://www.nature.com/articles/s41586-023-06415-8).

    Note: the RF in RFDiffusion stands for RoseTTAFold, which you can read more on [here](https://www.science.org/doi/10.1126/science.abj8754).



## Visualization
  - **Chimera**: Best for generating final graphics for presentations, models alpha helix versus beta pleats. Chimera tutorials can be found [here](https://www.cgl.ucsf.edu/chimera/tutorials.html).
  - **VMD**: Best for modeling multiple docking poses and animating mechamisms. Many options for color/shape of models. VMD tutorials can be found [here](https://www.ks.uiuc.edu/Training/Tutorials/vmd/tutorial-html/).
