# 3. Protein Structure Prediction

Since a protein's entire role in the body is controlled by its structure, researchers have been studying the protein folding process and trying to simulate it computationally. Through such simulations, scientists can not only better understand what impacts the folding and structure of a protein, but also fold de novo (new) proteins and test their capabilities computationally before using resources in the wet lab. 

## Protein Folding Programs
  - [**AlphaFold2**](https://github.com/alyssablood/Perez-Lab-Intro-Tutorials/blob/main/5.%20Protein%20Interactions.md#alphafold-competitive-binding-assays-af-cba): Extremely popular tool for deriving a 3D structure from an amino acid sequence. Very high accuracy, industry standard. You can read the AlphaFold official paper [here](https://www.nature.com/articles/s41586-021-03819-2).
  - [**AlphaFold3**](https://github.com/alyssablood/Perez-Lab-Intro-Tutorials/blob/main/5.%20Protein%20Interactions.md#alphafold-competitive-binding-assays-af-cba): Expansion of AlphaFold2 that also analyzes interacions between molecules. This program won 2024 Nobel Prize in Chemistry. You can read the AlphaFold3 official paper [here](https://www.nature.com/articles/s41586-024-07487-w).
  - [**AMBER**](https://github.com/alyssablood/Perez-Lab-Intro-Tutorials/blob/main/6.%20Protein%20Dynamics.md#beyond-static-folds-ensembles-and-motion): Uses physics-based (p-chem) methods to apply [force fields](https://en.wikipedia.org/wiki/Force_field_(chemistry)) (vector equations) to molecules in order to predict folding, docking, and general behavior. Good for very specific results, binding energy and other outputs are considered reliable. AMBER's homepage and tutorials are linked [here](https://ambermd.org/index.php).

## How Do They Work 

While each program works differently, the concept of predicting protein structures can best be seen through the AlphaFold architecture -
![alphafold](./images/alphafold.png)

Like all neural-network-based AI programs, AlphaFold has an **Input**(protein sequence), a **Hidden Layer** where all the folding takes place, and an **Output** (protein structure).

In the Hidden Layer, the amino acid sequence is compared to other amino acid sequences in the Protein Data Bank (PDB), and a **multiple sequence alignment (MSA)** is performed to see what amino acids are conserved (the same across multiple sequences) as well as the general patterns displayed. At the same time, the input amino acid sequence is also searched for structures in the PDB, and the same alignment method is used to compare similar structures and find patterns. This is shown in the "B. database search" in the image above. These patterns are then fed to the Evoformer, combined, and used to generate an initial structure, as shown in the "C. prediction model" in the image above. However, this structure is very poor and usually doesn't represent a structure that would exist in the real world. So, the prediction process is recycled three times, yielding the final protein structure.

## AlphaFold Tutorial

### [Docking](https://github.com/alyssablood/Perez-Lab-Intro-Tutorials/blob/main/5.%20Protein%20Interactions.md#docking-methods)
  - Docking programs can be very similar to protein dynamics, however they are not capable of predicting mechanisms as they do not consider time-dependant factors. They are used primarily to predict final docked conformations.
  - [**AutoDock**](https://github.com/alyssablood/Perez-Lab-Intro-Tutorials/blob/main/5.%20Protein%20Interactions.md#docking-methods): Predicts how ligand and protein interact, and what binding conformation would occur. P-chem methods are NOT used, therefore specific numbers are considered unreliable. Good for preliminary testing to get shape and quantity of docked poses. An overview of the AutoDock suite's capabilities can be found [here](https://pmc.ncbi.nlm.nih.gov/articles/PMC4868550/).

## [Protein Design](https://github.com/alyssablood/Perez-Lab-Intro-Tutorials/blob/main/4.%20Protein%20Design.md#4-protein-design)
  - [**BindCraft**](https://github.com/alyssablood/Perez-Lab-Intro-Tutorials/blob/main/4.%20Protein%20Design.md#bindcraft-and-other-emerging-methods): Good for desiging a new binder custom fit to a target protein. It is generally very accurate, but follows strict garbage in garbage out policy. Input protein must be high accuracy, and runs take a relatively long time. You can read the BindCraft official paper [here](https://www.nature.com/articles/s41586-025-09429-6). Click [here](https://github.com/alyssablood/Perez-Lab-Intro-Tutorials/blob/main/8.%20Community%20Challenges%3A%20unbiased%20assessment%20of%20methods.md#igem-projects-on-campus) to see how UF's undergraduate synthetic biology team (iGEM) is used BindCraft for deep space travel research!
  - [**ProteinMPNN**](https://github.com/alyssablood/Perez-Lab-Intro-Tutorials/blob/main/4.%20Protein%20Design.md#rfdiffusion-and-proteinmpnn): Optimizes amino acid sequences for most stable protein backbones. Good for designing new target proteins. You can read a paper introducing ProteinMPNN [here](https://www.science.org/doi/10.1126/science.add2187).
  - [**RFDiffusion**](https://github.com/alyssablood/Perez-Lab-Intro-Tutorials/blob/main/4.%20Protein%20Design.md#rfdiffusion-and-proteinmpnn): Generates new protein backbones using diffusion models. Adds in noise to allow for flexibility in folding. Results MUST be checked carefully. It is very easy to add too much noise and generate garbage outputs. You can read RFDiffusion's official paper [here](https://www.nature.com/articles/s41586-023-06415-8).

    Note: the RF in RFDiffusion stands for RoseTTAFold, which you can read more on [here](https://www.science.org/doi/10.1126/science.abj8754).



## Visualization
  - **Chimera**: Best for generating final graphics for presentations, models alpha helix versus beta pleats. Chimera tutorials can be found [here](https://www.cgl.ucsf.edu/chimera/tutorials.html).
  - **VMD**: Best for modeling multiple docking poses and animating mechamisms. Many options for color/shape of models. VMD tutorials can be found [here](https://www.ks.uiuc.edu/Training/Tutorials/vmd/tutorial-html/).
